pkg/virt-api/webhooks/validating-webhook/admitters/pod-eviction-admitter.go

```diff
@@ -34,6 +36,18 @@ func NewPodEvictionAdmitter(clusterConfig *virtconfig.ClusterConfig, kubeClient
 	}
 }
 
+func isDryRun(ar *admissionv1.AdmissionReview) bool {
+	dryRun := ar.Request.DryRun != nil && *ar.Request.DryRun == true
+
+	if !dryRun {
+		evictionObject := policyv1.Eviction{}
+		if err := json.Unmarshal(ar.Request.Object.Raw, &evictionObject); err == nil && len(evictionObject.DeleteOptions.DryRun) > 0 {
```

Comment:
Looking into it, the issue is caused by nil deleteOptions:
```suggestion
		if err := json.Unmarshal(ar.Request.Object.Raw, &evictionObject); err == nil && evictionObject.DeleteOptions != nil && len(evictionObject.DeleteOptions.DryRun) > 0 {
```
**BUT**  we here should also cover evictv1beta1 requests.
In case of evictv1beta1 requests with dryrun we still fall in the case where we don't satisfy the dryrun option.

Our unit test does not check the opts and currently the `PatchAction` does not keep track of them.
IMO we have 2 options here:
1. Extend the PatchAction to keep track of the options and check them in the test
2. Extend the e2e tests coverage with both evictv1 and evictv1beta1

Reply:
I've fixed the nil pointer panic.

I've also disabled this unit as to properly unit test this we need to bump the k8s version of the fake client to v0.31, and right now this is not trivial because to do that bump we need to use openapiv3 which is non trivial change.

For the moment I'd like to have this fixed even if untested, bump k8s to v0.31 and then re-enable this test.